health/reducers/multistream.go

```diff
@@ -46,19 +46,36 @@ func (t MultistreamReducer) Reduce(current *data.HealthStatus, _ interface{}, ev
 
 	multistream := current.MultistreamCopy()
 	multistream, idx := findOrCreateMultistreamStatus(multistream, target)
-	if status := connectedStatusFromEvent(evt); status != nil {
-		currConnected := multistream[idx].Connected
+	currConnected := multistream[idx].Connected
+
+	switch evt.Event {
+	case "multistream.connected":
+		status := true
+		multistream[idx] = &data.MultistreamStatus{
+			Target:    target,
+			Connected: data.NewCondition("", ts, &status, currConnected),
+		}
+	case "multistream.error":
+		status := false
 		multistream[idx] = &data.MultistreamStatus{
 			Target:    target,
-			Connected: data.NewCondition("", ts, status, currConnected),
+			Connected: data.NewCondition("", ts, &status, currConnected),
 		}
+	case "multistream.disconnected":
+		multistream = append(multistream[:idx], multistream[idx+1:]...)
```

Comment:
An alternative instead of removing could be to create a separate condition in the `MultistreamStatus` struct which is `Error(ed)`, so we can differentiate targets that we have never heard about from targets that we have heard of but have been disabled or disconnected on their own after some time. 

Would only do this if we think there's any improvement in the processing of this data, like showing any info more accurately in the dashboard idk. Otherwise might be overengineering. Raising here for consideration anyway!

Reply:
IMO disconnected targets should not differ in the health stats from the ones that were never started. I think that it's enough to have these 3 states I mentioned in the comment above.